### DISCUSSION TITLE
ESP32 ability to repurpose the programming UART

### DISCUSSION THREAD

**Discussion by SpazCode153** [nanoframework/Home#1151] [Q&A] [answered]
I'm not sure if this question has been asked before, I did a search and couldn't find an answer. The first UART(UART0) on the ESP32 is used for programming and debugging with the nanoframework. Is it possible to reuse the first UART for communication with another device while the program is executing? I know this will remove the ability to debug the program but is it possible? I am unfortunately using the other 2 UARTs for other purposes and would like to use the first UART.

**Comment by Ellerbach** [comment, marked as answer]
No, it's not possible. The question has been asked in our Discord channel. Check the thread starting here: https://discordapp.com/channels/478725473862549535/481782754674212867/1016999977374994452

There is a need to have this UART0 used. But if you're building your own image, you can adjust things. Some elements as well in the thread I pointed.

**Reply by SpazCode153** [reply]
Thanks @Ellerbach, on a side note, do you know if UART1 is dedicated for anything specific? I'm asking as I'm trying to create a SerialPort using COM1(UART1) but it fails with a ArugmentException.
